service/controller/v3/resource/deployment/deployment.go

```diff
@@ -21,53 +21,53 @@ func getDeploymentNames() []string {
 	}
 }
 
-func (r Resource) newDeployment(ctx context.Context, obj providerv1alpha1.AzureConfig, overwrites map[string]interface{}) (azureresource.Deployment, error) {
+func (r Resource) newDeployment(ctx context.Context, customObject providerv1alpha1.AzureConfig, overwrites map[string]interface{}) (azureresource.Deployment, error) {
```

Comment:
We call it `customObject` like everywhere so there is no reason to do it differently here.